src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -253,7 +253,25 @@ public sealed class ActivityListener : IDisposable
         public System.Diagnostics.SampleActivity<string>? SampleUsingParentId { get { throw null; } set { throw null; } }
         public System.Diagnostics.SampleActivity<ActivityContext>? Sample { get { throw null; } set { throw null; } }
         public void Dispose() { throw null; }
-   }
+    }
+
+    public delegate bool PropagatorGetterCallback(object carrier, string fieldName, out string? value);
+
+    public abstract class TextMapPropagator
+    {
+        public abstract System.Collections.Generic.IEnumerable<string> Fields { get; }
+        public abstract bool Inject(System.Diagnostics.Activity activity, object carrier, Action<object, string, string> setter);
+        public abstract bool Inject(System.Diagnostics.ActivityContext context, object carrier, Action<object, string, string> setter);
+        public abstract bool Inject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>> baggage, object carrier, Action<object, string, string> setter);
+        public abstract bool Extract(object carrier, string fieldName, PropagatorGetterCallback getter, out string? value);
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out System.Diagnostics.ActivityContext context);
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>>? baggage);
+        public static TextMapPropagator DefaultPropagator { get; set; }
+        public static TextMapPropagator CreateLegacyPropagator() { throw null; }
+        public static TextMapPropagator CreatePassThroughPropagator() { throw null; }
+        public static TextMapPropagator CreateOutputSuppressionPropagator() { throw null; }
+        public static TextMapPropagator CreateW3CPropagator() { throw null; }
```

Comment:
As an implementation detail, these should be caching a single instance IMO (each is allocating a HashSet internally so it's not just an empty object)

Reply:
Yes, this is implementation details. We can cache the instances we return if needed. I didn't bother in the protype to go to that level.